internal/admission/handler.go

```diff
@@ -242,7 +243,14 @@ func (h RequestHandler) handleSecret(
 	if err != nil {
 		return nil, err
 	}
-	if _, ok := secret.Data["kongCredType"]; !ok {
+	// TODO so long as we still handle the deprecated field, this has to remain
+	// once the deprecated field is removed, we must replace this with a label filter in the webhook itself
+	// https://github.com/Kong/kubernetes-ingress-controller/issues/4853
+
+	_, labelOk := secret.Labels[labels.LabelPrefix+labels.CredentialKey]
+	_, fieldOk := secret.Data["kongCredType"]
+
+	if !(labelOk || fieldOk) {
 		// secret does not look like a credential resource in Kong
 		return responseBuilder.Allowed(true).Build(), nil
 	}
```

Comment:
I think this logic should belong to `KongHTTPValidator.ValidateCredential` and we should have a unit test covering credentials validation for it.